src/plugins/JSTUN/JSTUN.java

```diff
@@ -25,292 +29,395 @@
-		return;
-	}
-
-	public void runPlugin(PluginRespirator pr) {
-		this.pr = pr;
-	}
-
-	public String handleHTTPGet(HTTPRequest request) throws PluginHTTPException {
-		PageNode page = pr.getPageMaker().getPageNode("JSTUN plugin", false, null);
-		HTMLNode pageNode = page.outer;
-		HTMLNode contentNode = page.content;
-
-		if(reportedData == null) {
-			if(hasRunTestBeenCalled) {
-				HTMLNode jSTUNReportInfobox = contentNode.addChild("div", "class", "infobox infobox-warning");
-				HTMLNode jSTUNReportInfoboxHeader = jSTUNReportInfobox.addChild("div", "class", "infobox-header");
-				HTMLNode jSTUNReportInfoboxContent = jSTUNReportInfobox.addChild("div", "class", "infobox-content");
-
-				jSTUNReportInfoboxHeader.addChild("#", "JSTUN detection report");
-
-				jSTUNReportInfoboxContent.addChild("#", "The plugin hasn't managed to contact any server yet.");
-			} else {
-				HTMLNode jSTUNReportInfobox = contentNode.addChild("div", "class", "infobox infobox-normal");
-				HTMLNode jSTUNReportInfoboxHeader = jSTUNReportInfobox.addChild("div", "class", "infobox-header");
-				HTMLNode jSTUNReportInfoboxContent = jSTUNReportInfobox.addChild("div", "class", "infobox-content");
-
-				jSTUNReportInfoboxHeader.addChild("#", "JSTUN detection report");
-
-				jSTUNReportInfoboxContent.addChild("#", "There is no need for the plugin to determine your ip address: the node knows it.");
-			}
-		} else {
-			HTMLNode jSTUNReportErrorInfobox = contentNode.addChild("div", "class", "infobox infobox-normal");
-			HTMLNode jSTUNReportInfoboxHeader = jSTUNReportErrorInfobox.addChild("div", "class", "infobox-header");
-			HTMLNode jSTUNReportInfoboxContent = jSTUNReportErrorInfobox.addChild("div", "class", "infobox-content");
-
-			jSTUNReportInfoboxHeader.addChild("#", "JSTUN detection report");
-
-			jSTUNReportInfoboxContent.addChild("#", "The plugin has reported the following data to the node:");
-			HTMLNode data = jSTUNReportInfoboxContent.addChild("pre");
-			data.addChild("#", reportedData.toString());
-		}
-		return pageNode.generate();
-	}
-
-	public String handleHTTPPost(HTTPRequest request)
-			throws PluginHTTPException {
-		return null;
-	}
-
-	public String handleHTTPPut(HTTPRequest request) throws PluginHTTPException {
-		return null;
-	}
-
-	public String getVersion() {
-		return "1.4";
-	}
-
-	public long getRealVersion() {
-		return 5;
-	}
+//threadless in the sense that it doesn't need a thread running all the time.
+//but getAddress() can and will block!
+public class JSTUN
+        implements FredPlugin, FredPluginIPDetector, FredPluginThreadless, FredPluginHTTP,
+                   FredPluginVersioned, FredPluginRealVersioned {
+
+    // From http://code.google.com/p/natvpn/source/browse/trunk/stun_server_list
+    // TODO: Google STUN servers run on port 19302 instead of 3478. Is it worth supporting them?
+    static String[] publicSTUNServers = new String[] {
+        //J-  Tell Jindent to not format the following
```

Comment:
I believe when JIndents auto-formatting is manually overridden it places
markers like that to keep the auto-formatting tool from changing it again
next pass.

If you are going to process the whole repository via script then just
discard these. :-)
On Jul 29, 2015 10:10 PM, "Steve Dougherty" <notifications@github.com>
wrote:

> In src/plugins/JSTUN/JSTUN.java
> <https://github.com/freenet/plugin-JSTUN/pull/5#discussion_r35832606>:
>
> > -		return "1.4";
> > -	}
> > -
> > -	public long getRealVersion() {
> > -		return 5;
> > -	}
> > +//threadless in the sense that it doesn't need a thread running all the time.
> > +//but getAddress() can and will block!
> > +public class JSTUN
> > +        implements FredPlugin, FredPluginIPDetector, FredPluginThreadless, FredPluginHTTP,
> > +                   FredPluginVersioned, FredPluginRealVersioned {
> > +
> > +    // From http://code.google.com/p/natvpn/source/browse/trunk/stun_server_list
> > +    // TODO: Google STUN servers run on port 19302 instead of 3478. Is it worth supporting them?
> > +    static String[] publicSTUNServers = new String[] {
> > +        //J-  Tell Jindent to not format the following
>
> What's the reasoning for keeping JIndent directives like this?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/freenet/plugin-JSTUN/pull/5/files#r35832606>.
>